crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -299,16 +299,9 @@ impl SignedCheckpointSummary {
             .verify(&self.summary, self.auth_signature.authority)?;
 
         if let Some(contents) = contents {
-            let recomputed = CheckpointSummary::new(
-                self.summary.epoch,
-                *self.summary.sequence_number(),
-                contents,
-                self.summary.previous_digest,
-            );
-
             fp_ensure!(
-                recomputed == self.summary,
-                SuiError::from("Transaction digest mismatch")
+                contents.digest() == self.summary.content_digest,
```

Comment:
This is simpler indeed, but error prone (and the point I make in the other PR about verification). If we augment the checkpoint summary with more structures, we need to remember to also augment the check here in the new version of the code. However, in the old version where we use the constructor, this will automatically update the check here.